### ISSUE TITLE
Maldet service won't start (at startup)

### ISSUE THREAD

**Issue by Codmic** [magenx/Magento-Automated-Server-Configuration-from-MagenX#79]
Hi, 

First of all, this is a very nice installation script! I first tested it in Virtualbox and later on a VPS in a VMWare vSphere environment. 

The first time I tested it, I had some issues with Maldet / ClamAV. I thought it might be some incident and when I run the script on another server and fresh install of CentOS minimal the problems would be solved but that is unfortunately not the case.

Below some more details about the errors, I tried to find out hat the problems are but cannot find a solutions easily so maybe you can help me out and fix these issues in a next version of the script!

Everything else installed fine and seems to work perfectly!

```
[ ! ALERT ! ] - maldet.service failed to start on xxxxx.com 10.1.8.13
Status report for unit: maldet.service

â— maldet.service - Linux Malware Detect monitoring - maldet
   Loaded: loaded (/etc/systemd/system/maldet.service; enabled; vendor preset: disabled)
   Active: activating (auto-restart) (Result: resources) since Thu 2016-10-27 10:16:29 CEST; 6ms ago
  Process: 4539 ExecStart=/usr/local/maldetect/maldet --monitor /usr/local/maldetect/monitor_paths (code=exited, status=0/SUCCESS)

Oct 27 10:16:29 vps-xxxxx.com systemd[1]: Failed to start Linux Malware Detect monitoring - maldet.
Oct 27 10:16:29 vps-xxxxx.com systemd[1]: Unit maldet.service entered failed state.
Oct 27 10:16:29 vps-xxxxx.com systemd[1]: Triggering OnFailure= dependencies of maldet.service.
Oct 27 10:16:29 vps-xxxxx.com systemd[1]: maldet.service failed.
```

```
[ ! ALERT ! ] - maldet.service failed to start on xxxxx.com 10.1.8.13
Status report for unit: maldet.service

â— maldet.service - Linux Malware Detect monitoring - maldet
   Loaded: loaded (/etc/systemd/system/maldet.service; enabled; vendor preset: disabled)
   Active: activating (auto-restart) (Result: resources) since Thu 2016-10-27 10:15:58 CEST; 12ms ago
  Process: 4143 ExecStart=/usr/local/maldetect/maldet --monitor /usr/local/maldetect/monitor_paths (code=exited, status=0/SUCCESS)

Oct 27 10:15:58 vps-xxxxx.com systemd[1]: maldet.service failed.

```

And this:

```
OSSEC HIDS Notification.
2016 Oct 27 10:15:27

Received From: vps-staging->/var/log/messages
Rule: 1002 fired (level 2) -> "Unknown problem somewhere in the system."
Portion of the log(s):

Oct 27 10:15:27 vps-staging systemd: Failed to start Linux Malware Detect monitoring - maldet.
```

When I look in /var/boot.log I also see:
`[FAILED] Failed to start Generic clamav scanner daemon.`

When I stop Maldet using "service maldet stop" an restart the service with "service maldet start" I see:

`Job for maldet.service failed because a configured resource limit was exceeded. See "systemctl status maldet.service" and "journalctl -xe" for details.`

The systemctl output is:

```
Oct 27 11:03:01 vps-xxxxx.com systemd[1]: PID file /usr/local/maldetect/tmp/inotifywait.pid not readable (yet?) after start.
Oct 27 11:03:01 vps-xxxxx.com systemd[1]: Failed to start Linux Malware Detect monitoring - maldet.
Oct 27 11:03:01 vps-xxxxx.com systemd[1]: Unit maldet.service entered failed state.
Oct 27 11:03:01 vps-xxxxx.com systemd[1]: Triggering OnFailure= dependencies of maldet.service.
Oct 27 11:03:01 vps-xxxxx.com systemd[1]: maldet.service failed.
Oct 27 11:03:04 vps-xxxxx.com systemd[1]: Stopped Linux Malware Detect monitoring - maldet.
```

**Comment by mrlerch** [comment]
I am not an expert, but did you disable Selinux and restart before running the script? I may be way off, but here and there I get it right :)

Mr. L

**Comment by Codmic** [comment]
Hi, yes SELinux was not running, the script also warns you to disable selinux before you can run it, so unfortunately this is not the case.

**Comment by magenx** [comment]
maldet needs inotifywait process to be available, probably inotify is not running

**Comment by Codmic** [comment]
I can see inotify is running by using tis command:

`ps -p $(find /proc/*/fd/* -type l -lname 'anon_inode:inotify' -print 2> /dev/null | sed -e 's/^\/proc\/\([0-9]*\)\/.*/\1/')`

The output:
`  PID TTY      STAT   TIME COMMAND
    1 ?        Ss     1:13 /usr/lib/systemd/systemd --switched-root --system --deserialize 21
  526 ?        Ss     0:02 /usr/lib/systemd/systemd-udevd
  667 ?        Ssl    0:30 /usr/sbin/NetworkManager --no-daemon
  675 ?        Ssl    0:13 /usr/sbin/rsyslogd -n
  676 ?        Ss     1:10 /bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation
  691 ?        Ss     0:05 /usr/sbin/crond -n
  703 ?        Ssl    0:20 /usr/lib/polkit-1/polkitd --no-debug
24856 ?        SN     0:55 /bin/inotifywait -r --fromfile /usr/local/maldetect/sess/inotify.paths.24106 --exclude (^/var/tmp/mysql.sock$|^/tmp/mysql.sock$|^/var/cache/buagent/m
30581 ?        S      0:00 /usr/sbin/CROND -n`

I just rebooted the system and then I get these messages at startup. These messages repeat and repeat untill I run 'service maldet stop'. This is on both the Virtual box machine where I tested the installation script as on the VPS where I performed the same actions. It would be great if I could get maldet to run without any problems. 

-------------------------------------------------------------
Subject:
[ ! ALERT ! ] - maldet.service failed to start on xxxxxx 10.1.8.13

Message:

```
Status report for unit: maldet.service


â— maldet.service - Linux Malware Detect monitoring - maldet
   Loaded: loaded (/etc/systemd/system/maldet.service; enabled; vendor preset: disabled)
   Active: activating (auto-restart) (Result: resources) since Tue 2016-11-01 15:24:13 CET; 9ms ago
  Process: 4092 ExecStart=/usr/local/maldetect/maldet --monitor /usr/local/maldetect/monitor_paths (code=exited, status=0/SUCCESS)

Nov 01 15:24:13 vps-staging.xxxxxxict.com systemd[1]: maldet.service failed.
```

-------------------------------------------------------------

Then about 10 sec later:

Subject:
maldet.service failed to start on xxxxxx 10.1.8.13

Message:

```
Status report for unit: maldet.service

â— maldet.service - Linux Malware Detect monitoring - maldet
   Loaded: loaded (/etc/systemd/system/maldet.service; enabled; vendor preset: disabled)
   Active: activating (auto-restart) (Result: resources) since Tue 2016-11-01 15:24:24 CET; 7ms ago
  Process: 4228 ExecStart=/usr/local/maldetect/maldet --monitor /usr/local/maldetect/monitor_paths (code=exited, status=0/SUCCESS)

Nov 01 15:24:24 vps-staging.xxxxxx.com systemd[1]: maldet.service failed.
```

-------------------------------------------------------------

And an OSSEC message:

Subject: 
OSSEC Notification - vps-staging - Alert level 7

Message:

```
OSSEC HIDS Notification.
2016 Nov 01 15:24:03

Received From: vps-staging->/var/log/messages
Rule: 1002 fired (level 2) -> "Unknown problem somewhere in the system."
Portion of the log(s):

Nov  1 15:24:03 vps-staging systemd: Failed to start Linux Malware Detect monitoring - maldet.

--END OF NOTIFICATION

OSSEC HIDS Notification.
2016 Nov 01 15:24:03

Received From: vps-staging->/var/log/messages
Rule: 1002 fired (level 2) -> "Unknown problem somewhere in the system."
Portion of the log(s):

Nov  1 15:24:03 vps-staging systemd: Unit maldet.service entered failed state.
```

**Comment by Codmic** [comment]
If it is any help, I see this when I run grep 'maldet' messages in /var/log/

```
Nov  1 15:22:33 vps-staging systemd: Starting Linux Malware Detect monitoring - maldet...
Nov  1 15:22:33 vps-staging maldet: Linux Malware Detect v1.5
Nov  1 15:22:33 vps-staging maldet: (C) 2002-2016, R-fx Networks <proj@rfxn.com>
Nov  1 15:22:33 vps-staging maldet: (C) 2016, Ryan MacDonald <ryan@rfxn.com>
Nov  1 15:22:33 vps-staging maldet: This program may be freely redistributed under the terms of the GNU GPL v2
Nov  1 15:22:33 vps-staging maldet: maldet(928): {mon} added /home/moviparts to inotify monitoring array
Nov  1 15:22:33 vps-staging maldet: maldet(928): {mon} added /var/tmp/ to inotify monitoring array
Nov  1 15:22:33 vps-staging rc.local: maldet(950): {mon} added /home/moviparts to inotify monitoring array
Nov  1 15:22:33 vps-staging maldet: maldet(928): {mon} added /tmp/ to inotify monitoring array
Nov  1 15:22:33 vps-staging rc.local: maldet(950): {mon} added /var/tmp/ to inotify monitoring array
Nov  1 15:22:33 vps-staging rc.local: maldet(950): {mon} added /tmp/ to inotify monitoring array
Nov  1 15:22:33 vps-staging maldet: maldet(928): {mon} starting inotify process on 3 paths, this might take awhile...
Nov  1 15:22:33 vps-staging rc.local: maldet(950): {mon} starting inotify process on 3 paths, this might take awhile...
Nov  1 15:22:35 vps-staging rc.local: maldet(950): {mon} inotify startup successful (pid: 1826 1802)
Nov  1 15:22:35 vps-staging rc.local: maldet(950): {mon} inotify monitoring log: /usr/local/maldetect/logs/inotify_log
Nov  1 15:22:35 vps-staging maldet: maldet(928): {mon} inotify startup successful (pid: 1826 1802)
Nov  1 15:22:35 vps-staging maldet: maldet(928): {mon} inotify monitoring log: /usr/local/maldetect/logs/inotify_log
Nov  1 15:22:35 vps-staging systemd: Failed to read PID from file /usr/local/maldetect/tmp/inotifywait.pid: Invalid argument
```

After that this is beeing logged about every minute.

```
Nov  1 15:25:04 vps-staging systemd: maldet.service holdoff time over, scheduling restart.
Nov  1 15:25:04 vps-staging systemd: Starting Linux Malware Detect monitoring - maldet...
Nov  1 15:25:15 vps-staging maldet: Linux Malware Detect v1.5
Nov  1 15:25:15 vps-staging maldet: (C) 2002-2016, R-fx Networks <proj@rfxn.com>
Nov  1 15:25:15 vps-staging maldet: (C) 2016, Ryan MacDonald <ryan@rfxn.com>
Nov  1 15:25:15 vps-staging maldet: This program may be freely redistributed under the terms of the GNU GPL v2
Nov  1 15:25:15 vps-staging maldet: maldet(4725): {mon} existing inotify process detected (try -k): 1826
Nov  1 15:25:15 vps-staging systemd: PID file /usr/local/maldetect/tmp/inotifywait.pid not readable (yet?) after start.
Nov  1 15:25:15 vps-staging systemd: Failed to start Linux Malware Detect monitoring - maldet.
Nov  1 15:25:15 vps-staging systemd: Unit maldet.service entered failed state.
Nov  1 15:25:15 vps-staging systemd: Triggering OnFailure= dependencies of maldet.service.
Nov  1 15:25:15 vps-staging systemd: maldet.service failed.
Nov  1 15:25:17 vps-staging systemd: Stopped Linux Malware Detect monitoring - maldet.
```

There is indeed no inotifywait.pid in the mentioned tmp dir.

When I run:
`ps -p $(find /proc/*/fd/* -type l -lname 'anon_inode:inotify' -print 2> /dev/null | sed -e 's/^\/proc\/\([0-9]*\)\/.*/\1/')
`

I see:

```
  PID TTY      STAT   TIME COMMAND
    1 ?        Ss     0:40 /usr/lib/systemd/systemd --switched-root --system --deserialize 21
  526 ?        Ss     0:01 /usr/lib/systemd/systemd-udevd
  665 ?        Ssl    0:08 /usr/sbin/rsyslogd -n
  667 ?        Ssl    0:17 /usr/sbin/NetworkManager --no-daemon
  679 ?        Ss     0:39 /bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation
  693 ?        Ss     0:02 /usr/sbin/crond -n
  706 ?        Ssl    0:11 /usr/lib/polkit-1/polkitd --no-debug
 1826 ?        SN     0:02 /usr/bin/inotifywait -r --fromfile /usr/local/maldetect/sess/inotify.paths.950 --exclude (^/var/tmp/mysql.sock$|^/tmp/mysql.sock$|^/var/cache/buagent/md0.
26440 ?        S      0:00 /usr/sbin/CROND -n
```

**Comment by gladroger** [comment]
I got the same error on a new installed today. I get bombed by
Status report for unit: maldet.service

maldet.service - Linux Malware Detect monitoring - maldet
   Loaded: loaded (/etc/systemd/system/maldet.service; enabled; vendor preset: disabled)
   Active: activating (auto-restart) (Result: resources) since Wed 2017-01-04 16:16:31 CET; 7ms ago
  Process: 2078 ExecStart=/usr/local/maldetect/maldet --monitor /usr/local/maldetect/monitor_paths (code=exited, status=0/SUCCESS)

Jan 04 16:16:31 domain.com systemd[1]: Triggering OnFailure= dependencies of maldet.service.
Jan 04 16:16:31 domain.com systemd[1]: maldet.service failed.

**Comment by magenx** [comment]
yes, i see it too,
if you kill and restart inotifywait and maldet monitoring , it will work then.

**Comment by webshopessentials** [comment]
While trying to reproduce https://github.com/magenx/Magento-Automated-Server-Configuration-from-MagenX/issues/86 again, I ran into this issue. I've ran the magenx script from scratch on an new VPS.
After a reboot of my VPS maldet is not started, (enabled via systemctl). I would like it to be started without manual intervention as suggested by @magenx, but I suspect I have to wait for for Maldet version 1.6: https://github.com/rfxn/linux-malware-detect/issues/168 (Seems to be in progress for 1.6)
